### PR TITLE
[7.x] Make Str::startsWith return false when given a null value

### PR DESCRIPTION
**Pull request by SjorsO**
Currently, if you give `Str::startsWith` an array of needles where one of the needles is `null`, it will always return true.

```php
Str::startsWith('some string', null); // false
Str::startsWith('some string' [null]); // true
```

This PR makes `Str::startsWith` always return false when given a `null` value. 

`Str::endsWith` did not have this problem, I've added a test to assert that. I've also added a test with a case-sensitive value, because there weren't any of those yet.

### CODE REVIEW

**Comment by vlakoff** [comment]
While working on #33434, and looking at this PR, I noticed that currently, `Str::contains` produces (by chance) the expected results with haystack `null`, but produces warnings with haystack `[null]`:

```
Str::contains('haystack', null);   // false
Str::contains('', null);           // false
Str::contains(null, null);         // false

Str::contains('haystack', [null]); // produces warning
Str::contains('', [null]);         // produces warning
Str::contains(null, [null]);       // produces warning
```

<br>After #33434 is merged, I'd suggest replacing in `Str::startsWith`, `Str::endsWith` and `Str::contains` e.g.:
```
foreach ((array) $needles as $needle) {
    if ($needle !== '' && substr($haystack, -strlen($needle)) === (string) $needle) {
        return true;
    }
}
```
with:
```
foreach ((array) $needles as $needle) {
    $needle = (string) $needle; // cast once for good here
    if ($needle !== '' && substr($haystack, -strlen($needle)) === $needle) {
        return true;
    }
}
```
Which would eliminate these kinds of issues.

**Comment by vlakoff** [comment]
See https://github.com/laravel/framework/pull/37087#issuecomment-2670402271.

tl;dr: As intended by this PR, `startsWith()` and `endsWith()` support needles like `['foo', null, 'bar']`. I suggest extending this support to `contains()` for consistency.
